tracer/src/Datadog.Trace/Ci/GitCommandHelper.cs

```diff
@@ -0,0 +1,192 @@
+    {
+        TelemetryFactory.Metrics.RecordCountCIVisibilityGitCommand(ciVisibilityCommand);
+        try
+        {
+            var sw = System.Diagnostics.Stopwatch.StartNew();
+            var gitOutput = ProcessHelpers.RunCommand(
+                                new ProcessHelpers.Command(
+                                    "git",
+                                    arguments,
+                                    workingDirectory,
+                                    outputEncoding: Encoding.Default,
+                                    errorEncoding: Encoding.Default,
+                                    inputEncoding: Encoding.Default,
+                                    useWhereIsIfFileNotFound: true,
+                                    timeout: TimeSpan.FromMinutes(5)),
+                                input);
+            TelemetryFactory.Metrics.RecordDistributionCIVisibilityGitCommandMs(ciVisibilityCommand, sw.Elapsed.TotalMilliseconds);
+            if (gitOutput is null)
+            {
+                TelemetryFactory.Metrics.RecordCountCIVisibilityGitCommandErrors(ciVisibilityCommand, MetricTags.CIVisibilityExitCodes.Unknown);
+                Log.Warning("ITR: 'git {Arguments}' command is null", arguments);
+            }
+            else if (gitOutput.ExitCode != 0)
+            {
+                TelemetryFactory.Metrics.RecordCountCIVisibilityGitCommandErrors(MetricTags.CIVisibilityCommands.GetRepository, TelemetryHelper.GetTelemetryExitCodeFromExitCode(gitOutput.ExitCode));
+            }
+
+            if (Log.IsEnabled(Vendors.Serilog.Events.LogEventLevel.Debug))
+            {
+                Log.Debug("Git command : {Command}", $"git {arguments}");
+                Log.Debug("  exit code : {Output}", gitOutput?.ExitCode);
+                Log.Debug("     output : {Output}", gitOutput?.Output ?? "<NULL>");
+                if (gitOutput is not null && gitOutput.Error is { Length: > 0 } err)
+                {
+                    Log.Debug("     error  : {Error}", err);
+                }
+            }
+
+            return gitOutput;
+        }
+        catch (System.ComponentModel.Win32Exception ex)
+        {
+            Log.Warning(ex, "ITR: 'git {Arguments}' threw Win32Exception - git is likely not available", arguments);
+            TelemetryFactory.Metrics.RecordCountCIVisibilityGitCommandErrors(ciVisibilityCommand, MetricTags.CIVisibilityExitCodes.Missing);
+            return null;
+        }
+    }
+
+    public static FileCoverageInfo[] GetGitDiffFilesAndLines(string workingDirectory, string baseCommit, string? headCommit = null)
+    {
+        try
+        {
+            // Retrieve PR list of modified files
+            var arguments = $"diff -U0 --word-diff=porcelain {baseCommit}";
+            if (!string.IsNullOrEmpty(headCommit))
+            {
+                arguments += $" {headCommit}";
+            }
```

Comment:
mini nit:

```suggestion
            // Retrieve PR list of modified files
            var arguments = string.IsNullOrEmpty(headCommit) ? $"diff -U0 --word-diff=porcelain {baseCommit}" : $"diff -U0 --word-diff=porcelain {baseCommit} {headCommit}";
```

Minimal change to avoid the concat allocation